### PR TITLE
fix(libscap,libsinsp): correctly manage memfd in scapfiles

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

/area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

New `scap_fd_type` were introduced but their parsing logic was not added to `scap_savefile.c`, breaking captures. Moreover the correspondents were missing in libsinsp.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Andreagit97** [comment]
Great catch!

File: userspace/libscap/linux/scap_fds.c [hunk 1]
```diff
@@ -333,8 +333,15 @@ int32_t scap_fd_handle_regular_file(struct scap_proclist *proclist, char *fname,
 	}
 	else if(fdi->type == SCAP_FD_FILE_V2)
 	{
-		scap_fd_flags_file(fdi, procdir);
-		strlcpy(fdi->info.regularinfo.fname, link_name, sizeof(fdi->info.regularinfo.fname));
+		if (0 == strncmp(link_name, "/memfd:", strlen("/memfd:")))
```
Comment (FedeDP) [inline userspace/libscap/linux/scap_fds.c:336]: Question: why is this here and not above, where we do all the `else if` logic?

Comment (therealbobo) [reply userspace/libscap/linux/scap_fds.c:336]: That's a quirk! All the others file descriptors in /proc/pid/fd are links meanwhile the memfd file descriptors are actually files. With that being said, placing that logic above will result in a piece of code that will be never reached. 😄

File: test/e2e/tests/test_network/test_network.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def expected_events(origin: dict, destination: dict) -> list:
             "evt.type": "accept4",
             "fd.name": None,
             "proc.cmdline": "nginx",
-            "proc.exe": "nginx: master proces",
+            "proc.exe": "nginx: master process nginx -g daemon off;",
```
Comment (FedeDP) [inline test/e2e/tests/test_network/test_network.py:64]: @Andreagit97 how is this possible that we haven't caught this before?

Comment (therealbobo) [reply test/e2e/tests/test_network/test_network.py]: I realised this too late: that was an expected failure on the modern probe. I just removed the commit. I think we are good to go!

Comment (Andreagit97) [reply test/e2e/tests/test_network/test_network.py]: This is our man https://github.com/falcosecurity/libs/pull/976

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve
